### PR TITLE
linux_ck: init at 5.12.0

### PR DESCRIPTION
**Pull request by dan4ik605743**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128897` at a8a70f58 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>45 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/231f8f1e431f472aae6303efb022003c">linuxPackages_ck.acpi_call</a></li>
    <li><a href="https://gist.github.com/23d459805e00ee30ae2ed1b597ecedb0">linuxPackages_ck.akvcam</a></li>
    <li><a href="https://gist.github.com/bd81ad22097f101c9dde8946997ffb04">linuxPackages_ck.anbox</a></li>
    <li><a href="https://gist.github.com/0befed0bcc82dfe53e56a54e92858b89">linuxPackages_ck.apfs</a></li>
    <li><a href="https://gist.github.com/d8a721b5fb18d8fdd6e924b9743d8ff1">linuxPackages_ck.batman_adv</a></li>
    <li><a href="https://gist.github.com/52098dabd10c45bfc17558464f64e7c4">linuxPackages_ck.bcc</a></li>
    <li><a href="https://gist.github.com/5f8a7f60bb215751a5b1e6ef4ea74a7b">linuxPackages_ck.can-isotp</a></li>
    <li><a href="https://gist.github.com/a2c3e166c22dfe978a571257ab9017f2">linuxPackages_ck.cryptodev</a></li>
    <li><a href="https://gist.github.com/30b7f866c2e33ec6dc042fa5356a2c42">linuxPackages_ck.ddcci-driver</a></li>
    <li><a href="https://gist.github.com/6487c7f7cdb4de34be417fcfb8b30995">linuxPackages_ck.digimend</a></li>
    <li><a href="https://gist.github.com/2ef7a2f95fc7a07a31d40b8f76ba70e8">linuxPackages_ck.ena</a></li>
    <li><a href="https://gist.github.com/c294d772bec04b7e5004359b83fe0b8d">linuxPackages_ck.evdi</a></li>
    <li><a href="https://gist.github.com/561ec40dffed18f0c303962433ac707d">linuxPackages_ck.fwts-efi-runtime</a></li>
    <li><a href="https://gist.github.com/39bcff30d996f6b2fe3df762cfb67e64">linuxPackages_ck.gcadapter-oc-kmod</a></li>
    <li><a href="https://gist.github.com/e991367e02903bdd7ab0518aaa21e3d8">linuxPackages_ck.hid-nintendo</a></li>
    <li><a href="https://gist.github.com/b11a8d22940fd8c2debe28c81a269861">linuxPackages_ck.jool</a></li>
    <li><a href="https://gist.github.com/9ca8e75f27aaf54494a238eea267062e">linux_ck</a> (linuxPackages_ck.kernel)</li>
    <li><a href="https://gist.github.com/2a1548b9a9d57cce31d3cc5c601abb99">linuxPackages_ck.lttng-modules</a></li>
    <li><a href="https://gist.github.com/cf53552267756677a2d0b52d713c0f1e">linuxPackages_ck.mba6x_bl</a></li>
    <li><a href="https://gist.github.com/3ae29993caf9a872f2bf10027b08dfb6">linuxPackages_ck.mbp2018-bridge-drv</a></li>
    <li><a href="https://gist.github.com/d1d30d496d1e3963dddb96705dbb67f5">linuxPackages_ck.mwprocapture</a></li>
    <li><a href="https://gist.github.com/6482e6661fd16760f46008c5457cdb2b">linuxPackages_ck.mxu11x0</a></li>
    <li><a href="https://gist.github.com/4e801db72c1ba571df25e687678ae55b">linuxPackages_ck.netatop</a></li>
    <li><a href="https://gist.github.com/5baeb96b10bd8541d1866c766aa9603f">linuxPackages_ck.oci-seccomp-bpf-hook</a></li>
    <li><a href="https://gist.github.com/ae346a595924a4e23af60efddf22498a">linuxPackages_ck.odp-dpdk</a></li>
    <li><a href="https://gist.github.com/7cf26e85dbcc34d201a4c20751189ff8">linuxPackages_ck.openafs</a> (linuxPackages_ck.openafs_1_8)</li>
    <li><a href="https://gist.github.com/0fb2d83f7b1e8df292a5225fd48bbd45">linuxPackages_ck.openafs_1_9</a></li>
    <li><a href="https://gist.github.com/21d8a6b24bc38aeb55b12162724348d7">linuxPackages_ck.openrazer</a></li>
    <li><a href="https://gist.github.com/80d39683be3d81123fd849b61e0504b5">linuxPackages_ck.r8125</a></li>
    <li><a href="https://gist.github.com/86715d30ed40f1e39dc9249165afe887">linuxPackages_ck.r8168</a></li>
    <li><a href="https://gist.github.com/334ee4655fe975cf92371d4c26363bc8">linuxPackages_ck.rtl8192eu</a></li>
    <li><a href="https://gist.github.com/6b7d05a2b55313b41d7f7c263588e2e1">linuxPackages_ck.rtl8723bs</a></li>
    <li><a href="https://gist.github.com/e7c91e40dbb4868001ba21bc07afc238">linuxPackages_ck.rtl8812au</a></li>
    <li><a href="https://gist.github.com/f22eb2e3432e4152030345defdd7dc74">linuxPackages_ck.rtl8821cu</a></li>
    <li><a href="https://gist.github.com/e7dd4b956fab1fd1477c1e38c3c67eae">linuxPackages_ck.rtl88x2bu</a></li>
    <li><a href="https://gist.github.com/559a08287e98193f0590c5781c35895b">linuxPackages_ck.rtw88</a> (linuxPackages_ck.rtlwifi_new)</li>
    <li><a href="https://gist.github.com/e8ec57c0ed1a2d69d2786e8ca6a09116">linuxPackages_ck.sch_cake</a></li>
    <li><a href="https://gist.github.com/c0fe587512177c41ec7b11fce10046de">linuxPackages_ck.systemtap</a></li>
    <li><a href="https://gist.github.com/e0db1fc538d361a0d3e57e1f83d37472">linuxPackages_ck.tuxedo-keyboard</a></li>
    <li><a href="https://gist.github.com/4127cf3e9a95cd935a09de165c9c5bc7">linuxPackages_ck.v4l2loopback</a></li>
    <li><a href="https://gist.github.com/82509922ff749a33254671dfd97f5164">linuxPackages_ck.veikk-linux-driver</a></li>
    <li><a href="https://gist.github.com/0cfadc07b3727357fb2fbea4f2f8a775">linuxPackages_ck.vhba</a></li>
    <li><a href="https://gist.github.com/4b9db741eb1311bb74bfc30566be40a3">linuxPackages_ck.xpadneo</a></li>
    <li><a href="https://gist.github.com/fe43df2d4b21c23b625be60f4fe4f880">linuxPackages_ck.zfs</a> (linuxPackages_ck.zfsStable)</li>
    <li><a href="https://gist.github.com/db3873caef7d26262e2db14ec773de27">linuxPackages_ck.zfsUnstable</a></li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>linuxPackages_ck.dpdk-kmods</li>
    <li>linuxPackages_ck.rtl8821ce</li>
  </ul>
</details>
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>linuxPackages_ck.cpupower</li>
    <li>linuxPackages_ck.hyperv-daemons</li>
    <li>linuxPackages_ck.perf</li>
    <li>linuxPackages_ck.stdenv</li>
    <li>linuxPackages_ck.tmon</li>
    <li>linuxPackages_ck.usbip</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by dan4ik605743** [comment]
What to do with the kernel?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20850,6 +20850,11 @@ in
     ];
   };
 
+  linux_ck = callPackage ../os-specific/linux/kernel/linux-ck.nix {
+    kernelPatches = [
+    ];
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:20855]:
```suggestion
    kernelPatches = [ ];
```

**Comment by SuperSandro2000** [comment]
It is not great that so much fails to build with it.

**Comment by dan4ik605743** [comment]
> It is not great that so much fails to build with it.

Close PR?

**Pull request closed without merging**